Media 2020 chapter

bazzadp opened the pull request:
Closes #900 

It's time we finished out this chapter. So I've converted [the current draft](https://docs.google.com/document/d/1ip2At9j6ba17odQwBbXDVU7qiLnHRnX8DflAfdQNb9c/edit#) to markup and added figures where I saw fit.

I've staged a version here so you can see what it looks like:
https://20210117t090507-dot-webalmanac.uk.r.appspot.com/en/2020/media
(edit - updated link)

Please review.

bazzadp commented on src/config/2020.json:
```diff
@@ -571,6 +571,16 @@
       "github": "ericwbailey",
       "twitter": "ericwbailey"
     },
+    "eeeps": {
+      "name": "Eric Portis",
+      "teams": [
+        "authors"
+      ],
+      "avatar_url": "https://avatars1.githubusercontent.com/u/3441390?v=4&s=200",
+      "website": "https://ericportis.com/",
+      "github": "eeeps",
+      "twitter": "etportis"
+    },
```
@eeeps please confirm you're happy with this as you didn't have a contributor profile.

eeeps replied:
Works for me!

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -4,13 +4,14 @@ part_number: I
 description: Media chapter of the 2020 Web Almanac covering image file sizes and formats, responsive images, client hints, lazy loading, accessibility and video.
-authors: [tpiros, bseymour]
-reviewers: [nhoizey, zcorpan, colinbendell, dougsillars]
+authors: [tpiros, bseymour, eeeps]
+reviewers: [nhoizey, colinbendell, dougsillars]
```
@tpiros can you confirm these are correct contributors in rough order of contribution?

tpiros replied:
Confirmed and added the person missing

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -4,13 +4,14 @@ part_number: I
 translators: []
-#tpiros_bio: TODO
-#bseymour_bio: TODO
+tpiros_bio: Tamas Piros is a Developer Evangelist, Google Developer Expert and Technical Trainer.
+bseymour_bio: Ben Seymour is a Dynamic Media & Content Specialist at <a href="https://cloudinary.com/">Cloudinary</a>.
+eeeps_bio: Eric Portis is a Web Platform advocate at <a href="https://cloudinary.com/">Cloudinary</a>.
```
@tpiros , @bseymour , @eeeps I've added a basic bio here. Please update as you see fit.

tpiros replied:
I have updated mine.

tpiros replied:
@eeeps please make your update as well so that we close this 🥳

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
 #featured_stat_label_3: TODO
```
@tpiros can you suggest a featured quote (typically a few sentences) and 3 stats with labels to appear on the home page to promote this chapter.

This is what last years chapter had:

![2019 featured quote](https://user-images.githubusercontent.com/10931297/104482935-7382d280-55bf-11eb-9344-1f6703c781c1.png)

tpiros replied:
I'd suggest the following for the feat quote (Ben's image intro, modified slightly):

```
Images and videos offer us the potential for a most powerful pairing: instant communication, capable of triggering an innate emotional response. However, they require considered implementation techniques to avoid also being a burden to our web pages.
```

In terms of stats I'd add (if possible to find) **median size of home page with video**, **% of all sites using an image** and **most used image format**. @bazzadp would you be able to help us find this data, or is there no capacity to do this atm?

bazzadp replied:
Looks like we don't have any media sizes data. That seems a bit of a miss to me to be honest! Though I guess covered in the page weight chapter. Maybe should add a paragraph and link to that?

Anyway what do you think about the following for featured quote and stats?

```
featured_quote: Images and videos offer us the potential for a most powerful pairing&colon; instant communication, capable of triggering an innate emotional response. However, they require considered implementation techniques to avoid also being a burden to our web pages.
featured_stat_1: 84.64%
featured_stat_label_1: Usage of WebP in mobile <code>&lt;Picture&gt;</code> element
featured_stat_2: 40.26%
featured_stat_label_2: JPG images as a percentage of all images
featured_stat_3: 57.22%
featured_stat_label_3: Video elements on desktop with the autoplay attribute
```

It looks like this on the home page:

![Featured Quote](https://user-images.githubusercontent.com/10931297/104728302-9c75a580-572e-11eb-8188-6e796f624c25.png)

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+Of pages using picture for format switching, around 68% are offering a single type variation, in addition to the img src which acts as the default.
+
+{{ figure_markup(
+  image="picture-format-usage-by-type.png",
+  caption="Picture format usage by type.",
+  description="Bar chart showing picture format usage by image type. WebP is dominate usage used by 83.29% of picture elements on desktop (84.64% on mobile). Next is JPG used by 4.84% and 4.83% respectively, followed by PNG at 18.18% and 17.46%, Gif at 0.53% and 0.53% and AVIF doesn't register above 0 for either client.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLNnD9VNqXNxMu60VovxIEp_L6vmNo1oWt8-C18DOetXB3qIkee_-KjZwYYPIkkIM-7So-5wBwQ4QY/pubchart?oid=775091522&format=interactive",
+  sheets_gid="1719719920",
+  sql_file="picture_format_distribution.sql"
+  )
+}}
+
+{# TODO Analysts this doesn't sound right? Is this the format chosen? That would make sense as we crawl as Chrome so it will prefer WebP over others, but if not then I'm very dubious about these stats. Can we double check? #}
```
```suggestion
{{ figure_markup(
  image="picture-format-usage-by-type.png",
  caption="Picture format usage by type.",
  description="Bar chart showing picture format usage by image type. WebP is dominate usage used by 83.29% of picture elements on desktop (84.64% on mobile). Next is JPG used by 4.84% and 4.83% respectively, followed by PNG at 18.18% and 17.46%, Gif at 0.53% and 0.53% and AVIF doesn't register above 0 for either client.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLNnD9VNqXNxMu60VovxIEp_L6vmNo1oWt8-C18DOetXB3qIkee_-KjZwYYPIkkIM-7So-5wBwQ4QY/pubchart?oid=775091522&format=interactive",
  sheets_gid="1719719920",
  sql_file="picture_format_distribution.sql"
  )
}}

{# TODO Analysts this doesn't sound right? Is this the format chosen? That would make sense as we crawl as Chrome so it will prefer WebP over others, but if not then I'm very dubious about these stats. Can we double check? #}
```

@smatei can you look into this?

bazzadp replied:
@smatei / @tpiros I added this text below the graph:

```md
We see that WebP is the dominante usage across `<picture>` elements, followed by PNG and JPG is only 4.83% of `<picture>` usage.
 
<p class="note">Note our crawler crawls as Chrome which supports WebP, but if using anoither browser which does not support this then you will see different results.</p>
```

bazzadp commented on src/content/en/2020/media.md:
````diff
@@ -23,3 +24,439 @@ queries: 05_Media
+</picture>
+```
+
+### Effective leveraging of more effective image formats
+
+{# TODO (authors) - can we have a sentence here to avoid an empty heading? Or should we just remove this heading and promot the other two under this? #}
````
@tpiros

bazzadp replied:
I added this:

```
### Effective leveraging of image formats

Using the appropriate image format and the capabilities that format offers is critical to make effective use of media on web pages.
```

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+Likewise, on the video front, the vast majority of resources are sent in the lowest-common-denominator format: MP4. We are  mystified as to why the second most popular format is the [now-depreciated](https://blog.adobe.com/en/publish/2017/07/25/adobe-flash-update.html#gs.my93m2) SWF, and curious whether these are playable in any microbrowser.
+
+### Usage of CDNs vs Local Domain for storing & delivering media
+
+{# TODO (authors) - can we have a sentence here to avoid an empty heading? #}
```
@tpiros

tpiros replied:
done

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+With HTML5, amongst other things, semantic elements were added to the language. These semantic elements give semantic meaning to content found within an HTML page. One element is the figure element which can, optionally use a fig caption element as its child. The main, and key difference between using a “simple” image tag (img) with a paragraph to describe the image versus using figure using a fi ...
+
+{{ figure_markup(
+  image="figure-and-figcaption-usage-by-page.png",
+  caption="Figure and Figcaption usage by page.",
+  description="Bar chart showing ",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLNnD9VNqXNxMu60VovxIEp_L6vmNo1oWt8-C18DOetXB3qIkee_-KjZwYYPIkkIM-7So-5wBwQ4QY/pubchart?oid=605432940&format=interactive",
+  sheets_gid="2037389060"
+  )
+}}
+
+{# TODO (analyst) - what's the SQL file for this? #}
```
@smatei

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+There are two ways to use a `<video>`. You can either stick a single resource URL into the `src` attribute on the element itself, or give it any number of child `<source>` elements, which the browser peruses until it finds a source it thinks it can load. Our first query looks at how often we see each of these patterns across all sampled pages.
+
+{{ figure_markup(
+  image="video-usage-of-src-versus-source.png",
+  caption="Video usage of Src versus Source.",
+  description="Bar chart showing ",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTLNnD9VNqXNxMu60VovxIEp_L6vmNo1oWt8-C18DOetXB3qIkee_-KjZwYYPIkkIM-7So-5wBwQ4QY/pubchart?oid=2100955508&format=interactive",
+  sheets_gid="689453572"
+  )
+}}
+
+{# TODO (analyst) - what's the SQL file for this? #}
```
@smatei

bazzadp reviewed (commented):
Highlighting remaining TODOs

Also need descriptions for some of the charts and then an edit.

bazzadp commented on src/content/en/2020/media.md:
```diff
@@ -4,13 +4,14 @@ part_number: I
-#tpiros_bio: TODO
-#bseymour_bio: TODO
+tpiros_bio: Tamas Piros is a Developer Experience Engineer at <a href="https://cloudinary.com/">Cloudinary</a>, Google Developer Expert and Technical Instructor.
+bseymour_bio: Ben Seymour is a Dynamic Media & Content Specialist at <a href="https://cloudinary.com/">Cloudinary</a>, author of "Practical Responsive Images".
```
Shall we link your book @bseymour ?

```suggestion
bseymour_bio: Ben Seymour is a Dynamic Media & Content Specialist at <a href="https://cloudinary.com/">Cloudinary</a>, author of <a href="http://responsiveimag.es/">Practical Responsive Images</a>.
```

And shame on you for not being on HTTPS yet these days! 😁

tpiros commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+### SEO & Accessibility
+
+{# TODO (authors) - can we have a sentence here to avoid an empty heading? #}
```
done

tpiros commented on src/content/en/2020/media.md:
```diff
@@ -23,3 +24,439 @@ queries: 05_Media
+
+## Videos
+
+{# TODO (authors) - can we have a sentence here to avoid an empty heading? #}
```
done

bazzadp commented:
OK I've updated the latest version here: https://20210115t133421-dot-webalmanac.uk.r.appspot.com/en/2020/media

I think we're good to merge now, so if all three authors could go into the "Files changed" tab, and click on the "Review changes" green button and Approve this change so I know you're happy.

And if I hear nothing by Sunday will assume you've no major issues and merge anyway 😁

Thanks again all!!

eeeps commented:
Did a big round of edits. The remaining blocker to release, I think, is four places where there are missing charts. I'll try to find some time today to pull those together.

bazzadp commented:
> Did a big round of edits. The remaining blocker to release, I think, is four places where there are missing charts. I'll try to find some time today to pull those together.

Thanks @eeeps I add the graphs (in some case big figures where the graphs wouldn't add anything as explained fully in text).

I also noticed the `preconnect` stat was wrong. It seemed really low and out of whack with resource hints chapter so investigated and found the issue (bad regex). Corrected the query and reran the stats and no pretty close to Resource Hints (who don't use regex).

Latest version staged here: https://20210116t184407-dot-webalmanac.uk.r.appspot.com/en/2020/media (and updated link in first comment so people don't look at an old version).

bazzadp commented:
Btw I agree with you on use of picture. The stat measure `<picture>` elements with at least one format. But not sure how you could have no formats so isn’t that just a measure of `<picture>` usage? @smatei ?

eeeps commented:
Looks great! Thank you!!

Gave it another full read-through and made a couple of tiny fixes... and called out a couple of y axises that I think are mislabeled with some TODOs. Other than that, I think this looks good.

bazzadp commented:
> Gave it another full read-through and made a couple of tiny fixes... and called out a couple of y axises that I think are mislabeled with some TODOs. Other than that, I think this looks good.

I fixed one of the y-axis, but the others are % of pages, not % of picture elements so left them as was (though changed "percent" to "percentage".

Also did some other small edits.

Latest (hopefully final?) version here: https://20210117t090507-dot-webalmanac.uk.r.appspot.com/en/2020/media

Will be releasing this, this evening Ireland time.
